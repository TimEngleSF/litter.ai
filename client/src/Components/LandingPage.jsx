import React from 'react';
import { PiFlowerTulipThin } from 'react-icons/pi';
import { Link } from 'react-router-dom';
import '../css/Landing.css';

export default function LandingPage({ user }) {
  return (
    <>
      <div className="landing-container main-container">
        <div className="landing-top-section">
          <PiFlowerTulipThin className="landing-logo" />
          <div className="landing-text">
            <h1>Welcome to LitterSort</h1>
            <p>With just a photo, sort your garbage and save the world.</p>
          </div>
          <div className="landing-top-buttons lower-buttons">
            <Link to="/capture">
              <button type="button" className="landing-capture-button">
                Capture Picture
              </button>
            </Link>
          </div>
        </div>
        <div className="landing-bottom-section">
          <div className="landing-bottom-buttons lower-buttons">
            <Link to="/leaderboard">
              <button type="button" className="landing-secondary-button">
                Leaderboard
              </button>
            </Link>
            { user
              ? (
                <Link to="/profile">
                  <button type="button" className="landing-secondary-button">
                    Profile
                  </button>
                </Link>
              )
              : (
                <>
                  <Link to="/register">
                    <button type="button" className="landing-secondary-button">
                      Sign Up
                    </button>
                  </Link>
                  <div className="landing-login-text">
                    <p>Already have an account?</p>
                    <Link to="/login">
                      Log in
                    </Link>
                  </div>
                </>
              )}
          </div>
        </div>
      </div>
      <div className="landing-full-bg" />
    </>
  );
}
