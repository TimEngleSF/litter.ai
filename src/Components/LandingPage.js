import React from 'react';

function LandingPage () {
    return (
        <div>
            <h1>Welcome to Litter.ai</h1>
            <p>With just a click, sort your garbage and save the world</p>
        </div>
    )
}

export default LandingPage;