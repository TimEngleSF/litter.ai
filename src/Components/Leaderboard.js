import React, { useState, useEffect } from 'react';
import { userPictureData as pictureData, userData } from '../sampleLeaderboardData.js'

export default function Leaderboard() {
  const [leaderboardData, setLeaderboardData] = useState([]);

  const renderTable = (user, index) => (
    <tr>
      <td>{index + 1}</td>
      <td>{user.username}</td>
      <td>{user.totalUploads}</td>
    </tr>
  )

  useEffect(() => {
    pictureData.sort((a, b) => (b.totalUploads - a.totalUploads))
    setLeaderboardData(pictureData);
  }, [leaderboardData])

  return (
    <div className="leaderboard-container">
      <table>
        <tr>
          <th scope="col">Rank</th>
          <th scope="col">Name</th>
          <th scope="col">Total</th>
        </tr>
        {leaderboardData.map((user, index) => (renderTable(user, index)))}
      </table>
    </div>
  )
}