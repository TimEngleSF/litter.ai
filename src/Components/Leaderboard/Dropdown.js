import React from 'react';
import Select from 'react-select';

const options = [
  { value: 'total', label: 'Total' },
  { value: 'paper', label: 'Paper' },
  { value: 'metal', label: 'Metal' },
  { value: 'cardboard', label: 'Cardboard' },
  { value: 'trash', label: 'Trash' },
  { value: 'glass', label: 'Glass' },
  { value: 'plastic', label: 'Plastic' },
  { value: 'compost', label: 'Compost' },
  { value: 'others', label: 'Others' },
  { value: 'unknown', label: 'Unknown' },
]

export default function Dropdown({ setCategory }) {
  const handleChange = (option) => {
    setCategory(option.value);
  }

  return (
    <Select
      options={options}
      defaultValue={options[0]}
      onChange={handleChange}
      styles={
        {singleValue: (baseStyles) => ({
          ...baseStyles,
          backgroundColor: '#076B49',
          color: 'white',
        }),
        control: (baseStyles) => ({
          ...baseStyles,
          backgroundColor: '#076B49',
          border: 'none',
        })
        }
      }
    />
  )
}