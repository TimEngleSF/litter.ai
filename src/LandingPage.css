.landing-page-container {
    height: 100vh;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    text-align: center;
  }
  
  .landing-page-image {
    align-self: flex-start;
    max-width: 100%;
  }
  
  .landing-page-text {
    flex-grow: 1;
  }
  
  .landing-page-buttons {
    align-self: flex-end;
    width: 100%;
    margin-bottom: 100px;
    box-sizing: border-box;
  }

  .button {
    flex: 1;
    width: 100%;
    max-width: 150px; 
    padding: 8px; 
    text-align: center;
  }

  .button:hover {
    background-color: #076B49;
    color: #FFF
  }

