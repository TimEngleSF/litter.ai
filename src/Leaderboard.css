/* Color variables */
:root {
  --main-green: #076B49;
  --sec-green: #74CC67;
  --white: #FFF;
  --lb-gray: #E0E9E6;
}

.lb-container {
  height: 80vh;
  width: 100vw;
  max-width: 768px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.lb-container .lb-user-stats {
  width: 80%;
  height: 5vh;
  display: flex;
  align-items: center;
  justify-content: space-around;
  border: 0.2rem var(--main-green) solid;
  border-radius: 5rem;
  margin-bottom: 2rem;
}

.lb-container .lb-table {
  width: 95%;
  height: 60vh;
  border: 0.15rem var(--main-green) solid;
  border-spacing: inherit;
}

.lb-container .lb-table tr:nth-of-type(even) {
  background: var(--lb-gray);
}

.lb-container .lb-table th + th {
  border-left: 0.1rem var(--white)solid;
}

.lb-container .lb-table td + td {
  border-left: 0.1rem var(--main-green)solid;
}

.lb-container .lb-header {
  background: var(--main-green);
  color: var(--white);
}

.lb-container .lb-header-name {
  height: 2.5rem;
  text-align: left;
  padding-left: 1rem;
}

.lb-container .lb-name {
  text-align: left;
  padding-left: 1rem;
}
