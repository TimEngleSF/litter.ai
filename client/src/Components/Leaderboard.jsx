/* eslint-disable react/jsx-no-useless-fragment */
import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import Dropdown from './Dropdown';
import { fetchLeaderboardData } from '../utils/fetchUserData';
import '../css/Leaderboard.css';

export default function Leaderboard() {
  const [leaderboardData, setLeaderboardData] = useState([]);
  const [dropdownSelection, setDropdownSelection] = useState('total');
  const [userRank, setUserRank] = useState(null);
  const [userItemCount, setUserItemCount] = useState(null);

  // Creates a data row for each user with their rank, username, and total uploads
  const renderTable = (user, index) => (
    <tr key={index}>
      <td>{index + 1}</td>
      <td className="lb-name">{user.username}</td>
      <td>{user.itemCount}</td>
    </tr>
  );

  // Sorts users by total uploads
  useEffect(() => {
    // Data for the total uploads is a different link than the categories
    const path = dropdownSelection === 'total' ? 'leaderboard' : `leaderboard/${dropdownSelection}`;

    // Authenticates user and grabs user's leaderboard data
    const fetchData = async () => {
      const response = await fetchLeaderboardData(path);
      setLeaderboardData(response.leaderboard);
      setUserRank(response.userRank);
      setUserItemCount(response.userItemCount);
    };
    fetchData();
  }, [dropdownSelection]);

  return (
    <>
      { leaderboardData.length > 0 && (
        <div className="lb-container main-container">
          <h1>Leaderboard</h1>
          { userRank && (
          <div className="lb-user-stats">
            <h3>
              Your rank:&nbsp;
              {userRank > 0 ? userRank : 'No photos'}
            </h3>
            <h3>
              {dropdownSelection}
              :&nbsp;
              {userItemCount}
            </h3>
          </div>
          )}
          <table className="lb-table">
            <thead>
              <tr className="lb-header">
                <th scope="col">Rank</th>
                <th scope="col">Name</th>
                <th scope="col">
                  <Dropdown
                    setLBCategory={setDropdownSelection}
                    defaultValue={null}
                    aria-label="Dropdown"
                  />
                </th>
              </tr>
            </thead>
            <tbody>
              {leaderboardData.map((user, index) => (renderTable(user, index)))}
            </tbody>
          </table>
          <div className="lb-buttons lower-buttons">
            <Link to="/capture">
              <button type="button">
                Capture Picture
              </button>
            </Link>
            <Link to="/">
              <button className="button-home" type="button">
                Home
              </button>
            </Link>
          </div>
        </div>
      )}
    </>
  );
}
