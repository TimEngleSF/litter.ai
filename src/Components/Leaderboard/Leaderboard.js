import React, { useState, useEffect } from 'react';
import axios from 'axios';
import Dropdown from './Dropdown';
import '../../Leaderboard.css';

export default function Leaderboard() {
  const [leaderboardData, setLeaderboardData] = useState([]);
  const [categoryOpen, setCategoryOpen] = useState(false);
  const [dropdownSelection, setDropdownSelection] = useState('total');
  const [userRank, setUserRank] = useState(0);

  // Creates a data row for each user with their rank, username, and total uploads
  const renderTable = (user, index) => (
    <tr key={index}>
      <td>{index + 1}</td>
      <td className="lb-name">{user.username}</td>
      <td>{user.totalUploads}</td>
    </tr>
  )

  const handleCategoryDropdown = (e) => {
    e.preventDefault();
    setCategoryOpen(!categoryOpen)
  }

  // Sorts users by total uploads
  useEffect(() => {
    axios.get('http://localhost:3001/leaderboard')
      .then((response) => {
        setLeaderboardData(response.data.leaderboard);
        setUserRank(response.data.rank);
      })
      .catch((err) => console.log(err));
  }, [])


  return (
    <div className="lb-container">
      <Dropdown setCategory={setDropdownSelection} cat={dropdownSelection} />
      <div className="lb-user-stats">
        <h3>Your rank: 9999</h3>
        <h3>Total: 99999999</h3>
      </div>
      <table className="lb-table">
        <thead>
          <tr className="lb-header">
            <th scope="col">Rank</th>
            <th scope="col" className="lb-header-name">Name</th>
            <th scope="col">Total
              {!categoryOpen && (
                <button onClick={handleCategoryDropdown}>^</button>
              )}
            </th>
          </tr>
        </thead>
      <tbody>
        {leaderboardData.map((user, index) => (renderTable(user, index)))}
      </tbody>
      </table>
    </div>
  )
}