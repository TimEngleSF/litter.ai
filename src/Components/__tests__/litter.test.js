import LandingPage from '../LandingPage';
import Leaderboard from '../Leaderboard/Leaderboard.js'
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react'

// test('First snapshot test', () => {
//     const component=renderer.create(<LandingPage />);
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
// });

describe('Leaderboard component', () => {
    it('Leaderboard matches the current snapshot', () => {
        const tree = renderer.create(<Leaderboard />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});