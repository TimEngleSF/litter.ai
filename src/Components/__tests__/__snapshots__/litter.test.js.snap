// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First snapshot test 1`] = `
<div>
  <h1>
    Welcome to Litter.ai
  </h1>
  <p>
    With just a click, sort your garbage and save the world
  </p>
</div>
`;

exports[`Leaderboard component Leaderboard matches the current snapshot 1`] = `
<div
  className="lb-container"
>
  <div
    className="lb-user-stats"
  >
    <h3>
      Your rank: 9999
    </h3>
    <h3>
      Total: 99999999
    </h3>
  </div>
  <table
    className="lb-table"
  >
    <thead>
      <tr
        className="lb-header"
      >
        <th
          scope="col"
        >
          Rank
        </th>
        <th
          scope="col"
        >
          Name
        </th>
        <th
          scope="col"
        >
          <div
            className=" css-b62m3t-container"
            onKeyDown={[Function]}
          >
            <span
              className="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
            />
            <div
              className=" css-jc6pza-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-1fdsijx-ValueContainer"
              >
                <div
                  className=" css-ayhps2-singleValue"
                >
                  Total
                </div>
                <input
                  aria-autocomplete="list"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-readonly={true}
                  className="css-mohuvp-dummyInput-DummyInput"
                  disabled={false}
                  id="react-select-2-input"
                  inputMode="none"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  tabIndex={0}
                  value=""
                />
              </div>
              <div
                className=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  className=" css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className=" css-1s7cnte-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
